### PR TITLE
feat(api): disable wasm endpoints when `wasm=off`

### PR DESCRIPTION
**Pull request by locao**
### Summary

When wasm module is disabled in `kong.conf`, users should not interact with Admin API `filter-chains` endpoints. This changes disables all wasm endpoints, returning `400` on every request.

### Checklist

- [x] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* Added routes for `filter-chains`, `services` and `routes` entities, overwriting the default ones when wasm is disabled.
* Tests.

### Issue reference

KAG-1925

### CODE REVIEW

File: spec/02-integration/20-wasm/01-admin-api_spec.lua [hunk 1]
```diff
@@ -387,4 +388,162 @@ describe("WASMX admin API [#" .. strategy .. "]", function()
...
+
+    describe("POST", function()
+      it("returns 400", function()
+        local res = admin:post("/filter-chains", {
+          headers = { ["Content-Type"] = "application/json" },
+          body = {
+            filters = { { name = "tests" } },
+            service = { id = service.id },
+          },
+        })
```
Comment (flrgh) [inline spec/02-integration/20-wasm/01-admin-api_spec.lua:426]: nit: you could update this to use the `json()` helper that the other test cases are using:

```suggestion
        local res = admin:post("/filter-chains", json {
          filters = { { name = "tests" } },
          service = { id = service.id },
        })
```

File: spec/02-integration/20-wasm/01-admin-api_spec.lua [hunk 2]
```diff
@@ -387,4 +388,162 @@ describe("WASMX admin API [#" .. strategy .. "]", function()
...
+    it("POST returns 400", function()
+      assert.response(
+        admin:post(path), {
+          headers = { ["Content-Type"] = "application/json" },
+          body = {
+            filters = { { name = "tests" } },
+            service = { id = service.id },
+          },
+        }
+      ).has.status(400)
```
Comment (flrgh) [inline spec/02-integration/20-wasm/01-admin-api_spec.lua:510]: I think this is a typo--your request params are passed to `assert.response()` and not `admin:post()`:

```suggestion
        admin:post(path, json {
          filters = { { name = "tests" } },
          service = { id = service.id },
        }
      ).has.status(400)
```

File: spec/02-integration/20-wasm/01-admin-api_spec.lua [hunk 3]
```diff
@@ -387,4 +388,162 @@ describe("WASMX admin API [#" .. strategy .. "]", function()
...
+    end)
+
+    it("PUT returns 400", function()
+      assert.response(
+        admin:put(path .. "/" .. utils.uuid()), {
```
Comment (flrgh) [inline spec/02-integration/20-wasm/01-admin-api_spec.lua:527]: nit: same as above re: `json()` helper

File: spec/02-integration/20-wasm/01-admin-api_spec.lua [hunk 4]
```diff
@@ -387,4 +388,162 @@ describe("WASMX admin API [#" .. strategy .. "]", function()
...
+
+    assert(helpers.start_kong({
+      database = strategy,
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      wasm = "off",
+    }))
+
+    admin = helpers.admin_client()
+  end)
+
```
Comment (flrgh) [inline spec/02-integration/20-wasm/01-admin-api_spec.lua:415]: Looks as though you're missing a `lazy_teardown()` to close the admin client and stop kong. I think this is why one of the other tests is failing due to not being able to start kong.

**Review by flrgh — CHANGES_REQUESTED** [review]
Looks pretty good to me, just needs some fixes to the tests.

File: spec/02-integration/20-wasm/01-admin-api_spec.lua (hunk 2 above)
Comment (locao) [reply spec/02-integration/20-wasm/01-admin-api_spec.lua:510]: Yes, lol, parenthesis closed too soon 🤦

File: spec/02-integration/20-wasm/01-admin-api_spec.lua (hunk 4 above)
Comment (locao) [reply spec/02-integration/20-wasm/01-admin-api_spec.lua:411]: Yep, good catch!

**Review by flrgh — APPROVED** [review]

**Pull request merged**